File: src/Solutions/Euler274.hs
module Solutions.Euler274 where
import Data.Maybe(catMaybes)
import Utils.NumberTheory(unPrime, invertMod, primesUpTo)

funF :: Int -> Int
funF = sum . catMaybes . map (invertMod 10 . unPrime) . primesUpTo . pred

euler274 :: IO String
euler274 = return $ show $ funF $ 10^7

File: rust/src/solutions/euler274.rs
use crate::utils::number_theory::invert_mod::InvertMod;

pub fn euler274() -> String {
    let bound = 10_000_000;
    primal::Sieve::new(bound)
        .primes_from(0)
        .take_while(|&n| n < bound)
        .filter_map(|p| 10i64.invert_mod(&(p as i64)))
        .sum::<i64>()
        .to_string()
}
